src/services/interfaces.ts

```diff
@@ -3,4 +3,8 @@ import type { JobRole } from "../models/job-role.js";
 export interface JobRoleService {
   getJobRoles(): JobRole[];
   getJobRoleById(id: number): JobRole | undefined;
+  getAllJobs(): Promise<JobRole[]>;
+  getJobById(id: number): Promise<JobRole>;
+  setBaseURL(baseURL: string): void;
+  getBaseURL(): string | undefined;
```

Comment:
not needed in the interface. controller will just call actions to get data.
While two services that implement the same interface will share the available methods, the constructors can be completely different - that's how you will be able to set the baseUrl (for the API 'flavour' of the service only)